### ISSUE TITLE
Meteor 1.5.1 module import self referencing files results in non-related error

### ISSUE THREAD

**Issue by Redroest** [meteor/meteor#9037]
OS: Fedora (linux) 25.
Meteor version: 1.5.1 (all packages up to date)

As a programmer I sometimes make mistakes. In this case, the mistake might cost me hours of frustration. The problem:

(example) When I create a blaze component in a file called main.js I use another import statement to import the corresponding template file: `import './main.html';`. Occasionally I mistake myself and reference the js file (that has the same name) instead: `import './main.js';`. Which (as a noob when it comes to the modules implementation of meteor) should theoretically cause an infinite loop.

I would expect the modules system to handle this gracefully by throwing an error along the lines of `"components/list/main.js" cannot reference itself`.

Instead. the system randomly throws an error in some other component that is not related at all to the (in this case the list) component that has an issue:
```
Uncaught TypeError: Cannot read property 'helpers' of undefined
```
Basically it could not detect the html file of that blaze component even if its correctly referenced. I'm absolutely sure that the reference is correct and removing the self-reference fixed the error. 

I suspect that it has to do with the fact that somewhere the html references for blaze simply break silently causing an error in the first js file that tries to reference a blaze template.

**Comment by abernix** [comment]
I'm not sure if I agree that the error `Uncaught TypeError: Cannot read property 'helpers' of undefined` is not related at all.  It's a result of the JavaScript code you have in `main.js` referring to `Template.<name>.helpers` without having `import`ed the `.html` file which contains the `<template name="<name>"></template>` template.  Thus, `Template.<name>` is `undefined` and you're trying to access the `helpers` property on an `undefined` object.

Doesn't the error also contain the exact line number of the problem, as shown in my image below?

![image](https://user-images.githubusercontent.com/841294/29611665-05239e16-8807-11e7-9258-40383d0bc7b0.png)

Clicking on `main.js:4` for me takes me right to the line at hand and I can see that `Template.<name>` is `undefined`.

> Basically it could not detect the html file of that blaze component even if its correctly referenced.

You seem to be saying that it was **not** correctly referenced.  If it's not working when it's correctly referenced (i.e. `import "./main.html"`) then that seems like a more serious problem.

**Comment by benjamn** [comment]
Self-imports are definitely allowed, so throwing an error in that case is not an option.

**Comment by abernix** [comment]
Also, what @benjamn stated above is what I had thought might be the case.  If you have any suggestions for fixing this while maintaining compliance with the specification, we'd be happy to hear it, but otherwise this isn't a bug (if anything this is a feature request anyway).

Thanks for bringing this up though!  If anything, this is a good issue for others to find via "Search" in the future. :smile:
